packages/bpk-component-graphic-promo/src/BpkGraphicPromo.js

```diff
@@ -0,0 +1,202 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkText from 'bpk-component-text';
+import BpkButton from 'bpk-component-button';
+import BpkCard from 'bpk-component-card';
+
+import TEXT_ALIGN from './BpkGraphicPromoTextAlign';
+import TEXT_COLORS from './BpkGraphicPromoTextColors';
+import STYLES from './BpkGraphicPromo.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  className: ?string,
+  kicker: ?string,
+  headline: string,
+  strapline: ?string,
+  image: string,
+  sponsor: ?{
+    label: string,
+    logo: string,
+    altText: string,
+  },
+  ctaText: string,
+  onClick: () => void,
+  invertVertically: boolean,
+  textAlign: TEXT_ALIGN,
+  textColor: TEXT_COLORS,
+};
+
+const constructAriaLabel = ({
+  ctaText,
+  headline,
+  kicker,
+  sponsor,
+  strapline,
+}: Props) => {
+  const text = [];
+  const addText = (value) => value && text.push(value);
+
+  if (sponsor) {
+    addText(sponsor.label);
+    addText(sponsor.altText);
+  } else {
+    addText(kicker);
+  }
+  addText(headline);
+  addText(strapline);
+  addText(ctaText);
+
+  return text.join('. ');
+};
+
+const BpkGraphicPromo = (props: Props) => {
+  const {
+    className,
+    ctaText,
+    headline,
+    image,
+    invertVertically,
+    kicker,
+    onClick,
+    sponsor,
+    strapline,
+    textAlign,
+    textColor,
+  } = props;
+
+  // FIXME: Use useCallback() here when React is updated.
+  const onClickWrapper = (event: React.MouseEvent<HTMLElement>) => {
+    event.stopPropagation();
+    onClick();
+  };
+
+  const cardClasses = [getClassName('bpk-graphic-promo')];
+  if (className) {
+    cardClasses.push(className);
+  }
+
+  const containerClasses = [
+    getClassName('bpk-graphic-promo__container'),
+    getClassName(`bpk-graphic-promo__container--${textAlign}`),
+  ];
+  if (invertVertically) {
+    containerClasses.push(
+      getClassName('bpk-graphic-promo__container--inverted'),
+    );
+  }
+
+  const getTextClasses = (baseClass: string) => {
+    const classes = [getClassName(baseClass)];
+    classes.push(getClassName(`${baseClass}--${textAlign}`));
+
+    return classes.join(' ');
+  };
+
+  return (
+    <BpkCard
+      className={cardClasses.join(' ')}
+      style={{ backgroundImage: `url(${image})`, color: textColor }}
+      onClick={onClickWrapper}
+      aria-label={constructAriaLabel(props)}
+      padded={false}
+    >
+      <div className={containerClasses.join(' ')} aria-hidden>
```

Comment:
If you do the above suggestions you simply then do
```suggestion
      <div className={containerClasses} aria-hidden>
```